src/btreemap/proptests.rs

```diff
@@ -1,17 +1,89 @@
 use crate::{
-    btreemap::test::{b, btree_test},
+    btreemap::{
+        test::{b, btree_test, make_memory},
+        BTreeMap,
+    },
     storable::Blob,
+    Memory,
 };
 use proptest::collection::btree_set as pset;
 use proptest::collection::vec as pvec;
 use proptest::prelude::*;
-use std::collections::BTreeSet;
+use std::collections::{BTreeMap as StdBTreeMap, BTreeSet};
 use test_strategy::proptest;
 
+#[derive(Debug, Clone)]
+enum Operation {
+    Insert { key: Vec<u8>, value: Vec<u8> },
+    Iter { from: usize, len: usize },
+    Get(usize),
+    Remove(usize),
+}
+
+// A custom strategy that gives unequal weights to the different operations.
+// Note that `Insert` has a higher weight than `Remove` so that, on average, BTreeMaps
+// are growing in size the more operations are executed.
+fn op_strategy() -> impl Strategy<Value = Operation> {
+    prop_oneof![
+        3 => (any::<Vec<u8>>(), any::<Vec<u8>>())
+            .prop_map(|(key, value)| Operation::Insert { key, value }),
+        1 => (any::<usize>(), any::<usize>())
+            .prop_map(|(from, len)| Operation::Iter { from, len }),
+        2 => (any::<usize>()).prop_map(Operation::Get),
+        1 => (any::<usize>()).prop_map(Operation::Remove),
+    ]
+}
+
 fn arb_blob() -> impl Strategy<Value = Blob<10>> {
     pvec(0..u8::MAX, 0..10).prop_map(|v| Blob::<10>::try_from(v.as_slice()).unwrap())
 }
 
+// Runs a comprehensive test for the major stable BTreeMap operations.
+// Results are validated against a standard BTreeMap.
+#[proptest(cases = 10)]
+fn comprehensive(#[strategy(pvec(op_strategy(), 100..5_000))] ops: Vec<Operation>) {
+    let mem = make_memory();
+    let mut btree = BTreeMap::new(mem);
+    let mut std_btree = StdBTreeMap::new();
+
+    // Execute all the operations, validating that the stable btreemap behaves similarly to a std
+    // btreemap.
+    for op in ops.into_iter() {
+        execute_operation(&mut std_btree, &mut btree, op);
```

Comment:
nit: I would rename this function. Maybe `validate_trees_after_operation`? Or something else that makes it obvious the purpose of the function is to verify the trees.

Reply:
Actually, the purpose of this method _is_ to execute an operation, so the current naming is accurate.